lib/APINotes/APINotesReader.cpp

```diff
@@ -1492,8 +1492,13 @@ APINotesReader::VersionedInfo<T>::VersionedInfo(
   Selected = Results.size();
 
   for (unsigned i = 0, n = Results.size(); i != n; ++i) {
-    if (Results[i].first == version) {
-      Selected = i;
+    if (version && Results[i].first >= version) {
+      // Pick the closest match. If the current version is "4", then entries for
+      // 4 are better than entries for 5, but both are valid.
+      if (Selected == Results.size() ||
+          Results[Selected].first > Results[i].first) {
+        Selected = i;
+      }
       break;
```

Comment:
Oops. This `break` is wrong. Either I need a test to catch this, or I should guarantee that the list is always sorted.